src/MolSim.cpp

```diff
@@ -73,25 +288,46 @@ int main(int argc, char *argsv[]) {
 }
 
 void calculateF() {
-  std::list<Particle>::iterator iterator;
-  iterator = particles.begin();
+  std::array<double, 3> sigma = {.0,.0,.0};
+  for (auto &p : particles) {
+    p.setOldF(p.getF());
+    p.setF(sigma);
+  }
+
+  for (auto &pair : particles.getParticlePairs()) {
+    Particle &p1 = pair.first.get();
+    Particle &p2 = pair.second.get();
 
-  for (auto &p1 : particles) {
-    for (auto &p2 : particles) {
-      // @TODO: insert calculation of forces here!
+    std::array<double, 3> sub = subtractVector(p2.getX(), p1.getX());
+    double normCubed = pow(magnitude(sub),3);
+
+    //prevent division by 0
+    if (normCubed == 0) {
```

Comment:
here instead of checking for `normCubed==0` I would recommend something like `normCubed < 1e-8` to be robust against floating point errors while calculating magnitude+pow.